src/java.base/share/classes/java/lang/invoke/LambdaMetafactory.java

```diff
@@ -229,6 +229,19 @@
  * used, but this is not compatible with some implementation techniques and
  * would complicate the work implementations must do.
  *
+ * <p>Uses besides evaluation of lambda expressions and method references are
+ * unintended.  These linkage methods may change their unspecified behaviors at
+ * any time to better suit the Java language features they were designed to
+ * support, and such changes may impact unintended uses.  Unintended uses of
+ * these linkage methods may lead to resource leaks, or other unspecified
+ * negative effects.
+ *
+ * @implNote In the reference implementation, the classes implementing the created
+ * function objects are strongly reachable from the defining class loader of the
+ * caller.  This technique reduces heap memory use, but it prevents the
+ * implementation class from unloading when the caller is a {@linkplain
+ * MethodHandles.Lookup.ClassOption#STRONG hidden class} and is unloaded.
```

Comment:
Sorry but, the text still doesn't seem right to me. The caller class is unloaded when its class loader is unloaded, which would also be when the implementation class is unloaded (its tied to the caller's class loader). Whether the caller is a hidden class (STRONG or not), should not matter... The point is that the _implementation class_  is a STRONG hidden class, so its lifetime gets tied to the caller's class loader.